### PR TITLE
[FIX] account,sale: restore record assignment notifications

### PR DESCRIPTION
**Pull request by yajo**
All accross Odoo, when a document is assigned to a salesperson, that person is notified via email.

However, the `account.move` model is inconsistent with that behavior, because the salesperson field name is `invoice_user_id` instead of just `user_id`.

With this fix:

- When creating an invoice from a sales order, the salesperson will get the assignment notification.
- When modifying the salesperson in the invoice form, the new salesperson will get the assignment notification.
- In both cases, the salesperson will get subscribed to the invoice.

This restores the behavior that `account.invoice` had in v12. For master, a better fix should be considered (a.k.a. simply removing the `invoice_user_id` field and using only `user_id` to avoid these kind of inconsistencies).

@Tecnativa TT29909

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/sale/models/sale.py [hunk 1]
```diff
@@ -544,6 +544,7 @@ def _prepare_invoice(self):
             'campaign_id': self.campaign_id.id,
             'medium_id': self.medium_id.id,
             'source_id': self.source_id.id,
+            'user_id': self.user_id and self.user_id.id,
```
Comment (pedrobaeza) [inline addons/sale/models/sale.py:547]: No need that repetition

```suggestion
            'user_id': self.user_id.id,
```

**Comment by pedrobaeza** [comment]
I see it a bit hackish, being `user_id` a related non stored field.

@oco-odoo what do you think about this?

**Comment by Yajo** [comment]
Well, it's hackish because I don't want to change the DB layout and because mail tracking works this way. See:
https://github.com/odoo/odoo/blob/652f5ff082e5b47f8aa2cc5e923a0ab5d230e78e/addons/mail/models/mail_thread.py#L2842-L2848

For master it can be really fixed, as explained in the description. If you see a better option for a stable release, please add your suggestion.

**Comment by Yajo** [comment]
Hello Odoo people! Any news on this?

**Comment by oco-odoo** [comment]
Could you explain the technical reason of this fix ? Is user_id getting a different value from invoice_user_id ? Why ?

**Comment by Yajo** [comment]
Do you mean why it works? As you see in https://github.com/odoo/odoo/issues/72939, the tracking stuff is handled in `.write()`. Since we weren't sending `user_id` in the call to `write()`, it didn't process tracking as it should. Above in https://github.com/odoo/odoo/pull/71328#issuecomment-849564204 you see that, if `user_id` is in the write dict, the assignment notification will be sent as expected.

**Review by oco-odoo — APPROVED** [review]
@robodoo r+

Ok for me. Indeed, a proper fix in master would be nice. Thanks for contributing :)

**Pull request closed without merging**